osidb/api_views.py

```diff
@@ -811,6 +819,49 @@ def destroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         return super().destroy(request, *args, **kwargs)
 
 
+class FlawAvailableView(
+    RudimentaryUserPathLoggingMixin,
+    APIView,
+):
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    http_method_names = ["get"]
+
+    def get(self, request, *args, **kwargs):
+        """
+        Report whether a flaw is available based on the following criteria:
+
+        1) Public / work on flaw is done: 200 status (yes)
+        2) Not public and work not done / flaw does not exist: 404 status (no)
+        """
+
+        pk = self.kwargs["flaw_id"]
+
+        """
+        The try-catch block is here so that the response is consistent
+        between criterion (2) and a flaw not being found.
+        """
+        try:
+            # Currently only allow search by CVE ID
+            if CVE_RE_STR.match(pk):
+                set_user_acls(settings.ALL_GROUPS)
+                flaw = get_object_or_404(Flaw, cve_id=pk)
+                self.check_object_permissions(self.request, flaw)
```

Comment:
This is not necessary and saves you the encompassing try/except:

```python
try:
    flaw = Flaw.objects.get(cve_id=pk)
except Flaw.DoesNotExist:
    return Response(status=HTTP_404_NOT_FOUND)
```

`get_object_or_404` is more useful for subviews e.g. `/flaws/<cve-id>/cvss-scores`, as you need the object to complete your task but is not the goal of the task itself.

FYI, while I'm aware we use `check_object_permissions` in OSIDB, we should avoid it as it's essentially useless: we don't use Django permissions in OSIDB, it's cargo cult at this point.